File: frontend/src/app/model/Order.ts
import {Symbol} from './Symbol';
import { OrderFilled } from './OrderFilled';

export class Order{
  orderId: number;
  userId: string;
  symbol: Symbol;
  orderSide: string;
  filledQuantity: number;
  filledTime: string;
  limitPrice: number;
  avgPrice: number;
  orderType: string;
  orderPlaced: string;
  orderStatus: string;
  newlyFilled: boolean;
  constructor(){
    this.orderId = null;
    this.symbol = null;
    this.orderSide = null;
    this.filledQuantity = 0;
    this.filledTime = null;
    this.limitPrice = 0;
    this.avgPrice = 0;
    this.orderType = 'LIMIT';
    this.orderPlaced = null;
    this.orderStatus = null;
    this.newlyFilled = false;
  }

  convertToOrder(o: OrderFilled):Order{
    this.symbol = new Symbol();

    this.orderId = o.orderId;
    this.symbol.symbol = o.symbol;
    this.symbol.name = o.name;
    this.symbol = this.symbol;
    this.symbol.name = o.name;
    this.orderSide = o.orderSide;
    this.filledQuantity = o.filledQuantity;

    this.filledTime = (o.filledTime=="null")?null:o.filledTime;
    this.limitPrice = o.limitPrice;
    this.avgPrice = o.limitPrice;
    this.orderType = o.orderType;
    this.orderPlaced = (o.orderPlaced=="null")?null:o.orderPlaced;
    this.orderStatus = o.orderStatus;
    this.newlyFilled = o.newlyFilled;
    return this;
  }
}

File: backend/src/main/java/com/canada/edu/stocktrading/model/Order.java
package com.canada.edu.stocktrading.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.Date;


@Entity
@Table(name = "orders")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id")
    private Integer orderId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id",
            foreignKey=@ForeignKey(name = "FK_USER_ORDER"))
    private UserEntity user;

   @ManyToOne(fetch = FetchType.LAZY)
   @JoinColumn(name = "symbol_id",
           foreignKey=@ForeignKey(name = "FK_SYMBOL_ORDER"))
   private Symbol symbol;

    @Enumerated(EnumType.STRING)
    @Column(name="order_side", nullable = false,length=10)
    private OrderSide orderSide;

    @Column(name="filled_quantity",nullable = false)
    private Integer filledQuantity;

    @Column(name="filled_time")
    @Temporal(TemporalType.TIMESTAMP)
    private Date filledTime;

    @Column(name="limit_price")
    private BigDecimal limitPrice;

    @Column(name="avg_price")
    private BigDecimal avgPrice;

    @Enumerated(EnumType.STRING)
    @Column(name="order_type", nullable = false, length=10)
    private OrderType orderType;

    @Column(name="order_placed",nullable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private Date orderPlaced;

    @Enumerated(EnumType.STRING)
    @Column(name="order_status", nullable=false, length=10)
    private OrderStatus orderStatus;

}


